wordpress-mobile/WordPress-iOS
# Improve text wrapping in FancyButtons

Fixes #16379. Also refs #16377. This PR pulls in an updated version of WordPressUI that improves automatic resizing of FancyButtons. It takes into account text wrapping when providing an intrinsic content size for the buttons.

| Before  | After  |
|---|---|
| ![](https://user-images.githubusercontent.com/66104034/116142705-76e62980-a697-11eb-9ce7-77264bbb91aa.jpeg) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-05-06 at 14 27 34](https://user-images.githubusercontent.com/4780/117414574-31cfbc00-af0f-11eb-80d7-1bc53468f475.png) |

When testing, I also ran through many other occurrences of FancyButton (and its subclass NUXButton) in the app to ensure nothing else appeared to be negatively affected by these changes:

|   |   |   |
|---|---|---|
| ![Simulator Screen Shot - iPhone 12 Pro - 2021-05-06 at 14 39 32](https://user-images.githubusercontent.com/4780/117414711-604d9700-af0f-11eb-8dde-429c66a037d0.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-05-06 at 14 39 39](https://user-images.githubusercontent.com/4780/117414716-617ec400-af0f-11eb-80bd-4fbfedf5ab3f.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-05-06 at 15 45 47](https://user-images.githubusercontent.com/4780/117414720-62175a80-af0f-11eb-8b4e-80b8a089c297.png) |
| ![Simulator Screen Shot - iPhone 12 Pro - 2021-05-06 at 14 34 23](https://user-images.githubusercontent.com/4780/117414763-6c395900-af0f-11eb-89a0-1ddba1605115.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-05-06 at 14 34 37](https://user-images.githubusercontent.com/4780/117414768-6cd1ef80-af0f-11eb-9fc2-f3c9a42f70b2.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-05-06 at 14 33 33](https://user-images.githubusercontent.com/4780/117414793-73f8fd80-af0f-11eb-86f2-d865530cfa93.png) |

**To test**

- Build and run
- Test on iPhone and iPad, with the default font size and also larger accessibility fonts
- To test the notifications alert, you can comment out the `guard` in `showNotificationPrimerAlertIfNeeded` in `NotificationsViewController`
- Check out a range of other FancyButtons – I just searched the source to see where they show up, and you can check the screenshots above.

Note that the buttons on the initial NUX screen still don't have enough vertical padding when the text wraps at large sizes, but this is due to existing extra constraints which we'll need to tweak separately in Authenticator.

## Regression Notes
1. Potential unintended areas of impact

Other FancyButton usages.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I tested many other areas where FancyButtons are used, including when text wraps, to ensure that they still look as expected.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
